src/collection.js

```diff
@@ -44,143 +109,422 @@ export default class Collection {
    * request’s success event, because the transaction may still fail after the
    * success event fires.
    *
-   * @param {String} mode  Transaction mode ("readwrite" or undefined)
+   * @param {String}      mode  Transaction mode ("readwrite" or undefined)
+   * @param {String|null} name  Store name (defaults to coll name)
    */
-  prepare(mode) {
-    const transaction = this._db.transaction([this.name], mode);
-    const store = transaction.objectStore(this.name);
+  prepare(mode=undefined, name=null) {
+    const storeName = name || this.name;
+    const transaction = this._db.transaction([storeName], mode);
+    const store = transaction.objectStore(storeName);
     return {transaction, store};
   }
 
+  /**
+   * Clears current collection.
+   *
+   * @return {Promise}
+   */
   clear() {
     return this.open().then(() => {
       return new Promise((resolve, reject) => {
         const {transaction, store} = this.prepare("readwrite");
         store.clear();
-        transaction.oncomplete = function(event) {
+        transaction.onerror = event => reject(new Error(event.target.error));
+        transaction.oncomplete = event => {
           resolve({
             data: [],
             permissions: {}
           });
         };
-        transaction.onerror = function(event) {
-          reject(new Error(event.target.error));
-        };
       });
-    });
+    }).catch(this._handleError("clear"));
   }
 
-  _create(record) {
+  /**
+   * Adds a record to the local database.
+   *
+   * @param  {Object} record
+   * @param  {Object} options
+   * @return {Promise}
+   */
+  create(record, options={synced: false}) {
     return this.open().then(() => {
+      if (typeof(record) !== "object")
+        return Promise.reject(new Error('Record is not an object.'));
       return new Promise((resolve, reject) => {
-        var {transaction, store} = this.prepare("readwrite");
-        var newRecord = Object.assign({}, record, {id: uuid4()});
+        const {transaction, store} = this.prepare("readwrite");
+        const newRecord = Object.assign({}, record, {
+          id:      options.synced ? record.id : uuid4(),
+          _status: options.synced ? "synced" : "created"
+        });
         store.add(newRecord);
-        transaction.oncomplete = function(event) {
+        transaction.onerror = event => reject(new Error(event.target.error));
+        transaction.oncomplete = event => {
           resolve({
             data: newRecord,
             permissions: {}
           });
         };
-        transaction.onerror = function(event) {
-          reject(new Error(event.target.error));
-        };
       });
-    });
+    }).catch(this._handleError("create"));
   }
 
-  _update(record) {
+  /**
+   * Updates a record from the local database.
+   *
+   * @param  {Object} record
+   * @param  {Object} options
+   * @return {Promise}
+   */
+  update(record, options={synced: false}) {
```

Comment:
Well, the `synced` name is the name of the status it reflects — I'm finding it pretty much explicit. I'll update jsdoc to better reflect its purpose if it's not clear.